common/dconfig/configupdater.go

```diff
@@ -0,0 +1,28 @@
+package dconfig
+
+import (
+	"github.com/uber/cherami-server/common"
+)
+
+// ConfigUpdater is a daemon that can be used to update the config based on information from some external source
+type (
+	ConfigUpdater interface {
```

Comment:
nit: ConfigUpdater is confusing IMO, given that the intention of your implementation is to detect / manage / orchestrate zone failover (based on info from external source). Would prefer a specific name, something along the lines of ZoneFailoverManager (or) ZoneFailoverDetector (or) better.